gdx/src/com/badlogic/gdx/graphics/g2d/TextureAtlas.java

```diff
@@ -30,170 +35,176 @@
-	static final String[] tuple = new String[4];
-
 	private final ObjectSet<Texture> textures = new ObjectSet(4);
 	private final Array<AtlasRegion> regions = new Array();
 
 	public static class TextureAtlasData {
-		public static class Page {
-			public final FileHandle textureFile;
-			public Texture texture;
-			public final float width, height;
-			public final boolean useMipMaps;
-			public final Format format;
-			public final TextureFilter minFilter;
-			public final TextureFilter magFilter;
-			public final TextureWrap uWrap;
-			public final TextureWrap vWrap;
-
-			public Page (FileHandle handle, float width, float height, boolean useMipMaps, Format format, TextureFilter minFilter,
-				TextureFilter magFilter, TextureWrap uWrap, TextureWrap vWrap) {
-				this.width = width;
-				this.height = height;
-				this.textureFile = handle;
-				this.useMipMaps = useMipMaps;
-				this.format = format;
-				this.minFilter = minFilter;
-				this.magFilter = magFilter;
-				this.uWrap = uWrap;
-				this.vWrap = vWrap;
-			}
-		}
-
-		public static class Region {
-			public Page page;
-			public int index;
-			public String name;
-			public float offsetX;
-			public float offsetY;
-			public int originalWidth;
-			public int originalHeight;
-			public boolean rotate;
-			public int degrees;
-			public int left;
-			public int top;
-			public int width;
-			public int height;
-			public boolean flip;
-			public int[] splits;
-			public int[] pads;
-		}
+		static final String[] entry = new String[5];
 
 		final Array<Page> pages = new Array();
 		final Array<Region> regions = new Array();
 
 		public TextureAtlasData (FileHandle packFile, FileHandle imagesDir, boolean flip) {
-			BufferedReader reader = new BufferedReader(new InputStreamReader(packFile.read()), 64);
+			ObjectMap<String, Field<Page>> pageFields = new ObjectMap(15, 0.99f); // Size needed to avoid collisions.
+			pageFields.put("size", new Field<Page>() {
+				public void parse (Page page) {
+					page.width = Integer.parseInt(entry[1]);
+					page.height = Integer.parseInt(entry[2]);
+				}
+			});
+			pageFields.put("format", new Field<Page>() {
+				public void parse (Page page) {
+					page.format = Format.valueOf(entry[1]);
+				}
+			});
+			pageFields.put("filter", new Field<Page>() {
+				public void parse (Page page) {
+					page.minFilter = TextureFilter.valueOf(entry[1]);
+					page.magFilter = TextureFilter.valueOf(entry[2]);
+				}
+			});
+			pageFields.put("repeat", new Field<Page>() {
+				public void parse (Page page) {
+					String direction = entry[1];
+					if (direction.equals("x"))
+						page.uWrap = Repeat;
+					else if (direction.equals("y"))
+						page.vWrap = Repeat;
+					else if (direction.equals("xy")) {
+						page.uWrap = Repeat;
+						page.vWrap = Repeat;
+					}
+				}
+			});
+			pageFields.put("pma", new Field<Page>() {
+				public void parse (Page page) {
+					page.pma = entry[1].equals("true");

```

Comment:
This line uses String.equals(), while below for rotate, String.equalsIgnoreCase() is used, both to compare against "true" and "false" (or "TRUE" and "FALSE"). I don't have a preference on which is correct, unless some older tools produced capitalized booleans (in which case both should ignore case).

Reply:
Good catch. I suppose we can use `equalsIgnoreCase` here and for repeat.